src/goto-programs/goto_program_template.h

```diff
@@ -807,7 +807,9 @@ struct const_target_hash_templatet
 {
   std::size_t operator()(
     const typename goto_program_templatet<codeT, guardT>::const_targett t) const
-  { return t->location_number; }
+  {
+    return reinterpret_cast<std::size_t>(&(*t));
+  }
```

Comment:
Why is this a better hash function, or rather: what bug is being "fixed" here?

Reply:
It appears that instruction location numbers were being modified while holding iterators to those same instructions in an unordered map. This breaks the map invariants.

Reply:
To be clear: this is the cause of the remaining bug in the slicer

Reply:
Oh, very nice catch (and likely a bug that I had introduced, sorry...). Would you mind giving a somewhat more detailed explanation in the commit message?